icu4c/source/test/cintltst/uformattedvaluetst.c

```diff
@@ -143,7 +143,7 @@ static void AssertAllPartsEqual(
 
     UErrorCode status = U_ZERO_ERROR;
 
-    char message[256];
+    char message[257];
```

Comment:
Instead of changing the size of the buffer and thereby pushing the underlying problem ahead until someone feels the need to change the size of it again in the future, I propose that you use `UPRV_LENGTHOF()` to replace the hardcoded buffer size whenever the buffer is being used, here and elsewhere in this PR.